### PR TITLE
GRAM:fix 4 slashes doc comment error

### PR DESCRIPTION
**Pull request by rrevenantt**
fix for #3379

### CODE REVIEW

File: src/main/grammars/RustLexer.flex [hunk 1]
```diff
@@ -228,7 +228,10 @@ OUTER_EOL_DOC = ({EOL_DOC_LINE}{EOL_WS})*{EOL_DOC_LINE}
   "/*"                            { yybegin(IN_BLOCK_COMMENT); yypushback(2); }
 
   {INNER_EOL_DOC}                 { return INNER_EOL_DOC_COMMENT; }
-  {OUTER_EOL_DOC}                 { return OUTER_EOL_DOC_COMMENT; }
+  {OUTER_EOL_DOC} / [^\/]         { //fix lookahead on EOF
```
Comment (mchernyavsky) [inline src/main/grammars/RustLexer.flex]: Is `\` needed here?

File: src/main/grammars/RustLexer.flex [hunk 2]
```diff
@@ -228,7 +228,10 @@ OUTER_EOL_DOC = ({EOL_DOC_LINE}{EOL_WS})*{EOL_DOC_LINE}
...
+                                    if (zzMarkedPos + 1 == zzBuffer.length())
```
Comment (mchernyavsky) [inline src/main/grammars/RustLexer.flex]: Can you explain what you are trying to fix here?
Now `///<EOF>` is not considered part of the doc comment:

Comment (rrevenantt) [reply src/main/grammars/RustLexer.flex]: Problem I am trying to fix is that you can't add <EOF> as accepted lookahead symbol which causes last symbol not to be included into doc comment token. So I just explicitly add last symbol to token. But apparently it is not enough. Then I suppose I need to write new lexer state for line doc comments.

**Comment by rrevenantt** [comment]
I have rewritten it as a separate lexer state. It should handle `<EOF>` better now.

**Comment by Undin** [comment]
bors r=mchernyavsky

**Comment by mchernyavsky** [comment]
bors retry

**Pull request merged**